### PR TITLE
Improve CTS build process with Intel compiler

### PR DESCRIPTION
**Pull request by bader**
- Use proper device triple
- Use device code split feature to improve JIT build time

### CODE REVIEW

File: cmake/FindIntel_SYCL.cmake [hunk 1]
```diff
@@ -19,7 +19,7 @@ add_library(INTEL_SYCL::Runtime INTERFACE IMPORTED GLOBAL)
 set_target_properties(INTEL_SYCL::Runtime PROPERTIES
     INTERFACE_LINK_LIBRARIES    OpenCL::OpenCL
     INTERFACE_COMPILE_OPTIONS   "-fsycl;-fsycl-targets=${INTEL_SYCL_TRIPLE};${INTEL_SYCL_FLAGS}"
-    INTERFACE_LINK_OPTIONS      "-fsycl;-fsycl-targets=${INTEL_SYCL_TRIPLE};${INTEL_SYCL_FLAGS}")
+    INTERFACE_LINK_OPTIONS      "-fsycl;-fsycl-device-code-split=per_kernel;-fsycl-targets=${INTEL_SYCL_TRIPLE};${INTEL_SYCL_FLAGS}")
```
Comment (vladimirlaz) [inline cmake/FindIntel_SYCL.cmake]: please change per_kernel to per_source as linking per_kernel cause too much memory consumption on vector_* tests.
Linking per source has reasonable time/memory consumption balance.

Comment (bader) [reply cmake/FindIntel_SYCL.cmake]: Done.

**Comment by bader** [comment]
@Ruyk, @ProGTX, @AerialMantis, could you take a look, please?

**Comment by Ruyk** [comment]
That looks fine to us

**Pull request merged**